src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,79 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<Integer> supplyAmounts = new ArrayList<>();
+        List<Integer> buyAmounts = new ArrayList<>();
+
+        readAndParseData(fromFileName, supplyAmounts, buyAmounts);
+
+        int[] statistics = calculateStatistics(supplyAmounts, buyAmounts);
+
+        writeResults(toFileName, statistics);
+    }
+
+    public void readAndParseData(String fromFileName, List<Integer> supplyAmounts,
+                                  List<Integer> buyAmounts) {
+        try (BufferedReader fromFile = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+
+            while ((line = fromFile.readLine()) != null) {
+                String[] values = line.split(",");
+
+                if (values.length == 2) {
+                    String operationType = values[0].trim();
+                    int amount = Integer.parseInt(values[1].trim());
+
+                    if (operationType.equalsIgnoreCase(SUPPLY)) {
+                        supplyAmounts.add(amount);
+                    } else if (operationType.equalsIgnoreCase(BUY)) {
+                        buyAmounts.add(amount);
+
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file", e);
+        }
+    }
+
+    public static int sumList(List<Integer> amounts) {
+        int sum = 0;
+        for (int amount : amounts) {
+            sum += amount;
+        }
+        return sum;
```

Comment:
The method `sumList` should be private if it's only intended for internal use within the class. Additionally, the method name could be more descriptive, for example, `calculateTotalAmount`.